src/AutoSchedule.cpp

```diff
@@ -679,14 +686,14 @@ map<string, Box> get_pipeline_bounds(DependenceAnalysis &analysis,
             int i;
             for (i = estimates.size() - 1; i >= 0; --i) {
                 const auto &est = estimates[i];
-                if (est.var == arg) {
+                if ((est.var == arg) && est.min.defined() && est.extent.defined()) {
                     Interval I = Interval(est.min, simplify(est.min + est.extent - 1));
                     pure_bounds.emplace(arg, I);
                     out_box.push_back(I);
                     break;
                 }
             }
-            internal_assert(i >= 0);
+            internal_assert(i >= 0) << "Could not find estimate for " << arg << "\n";
```

Comment:
Should this be user_assert?

Reply:
If check_estimates_on_outputs is working as expected, the assertion should never trigger within the get_pipeline_bounds